app/models/developer_key.rb

```diff
@@ -101,6 +101,14 @@ def account_name
     account.try(:name)
   end
 
+  def access_token_count
+    self.access_tokens.count
+  end
+
+  def last_used_at
+    self.access_tokens.order(last_used_at: :desc).first&.last_used_at
```

Comment:
Since we're ordering by `last_used` we'll need an index on it, also this can be made to not load the full ActiveRecord Object:

```ruby
def last_used
  self.access_tokens
    .order(last_used_at: :desc)
    .limit(1)
    .pluck(:last_used_at)
    .first
end
```

With an index Postgres will most likely just scan the index and pull the value from there w/o looking at the full table files when done this way.